packages/EasyDecision/src/Providers/AbstractMappingProvider.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyDecision\Providers;
+
+use EonX\EasyDecision\Exceptions\InvalidMappingException;
+use EonX\EasyDecision\Interfaces\MappingProviderInterface;
+
+abstract class AbstractMappingProvider implements MappingProviderInterface
+{
+    /**
+     * @var string[]
+     */
+    protected $typesMapping = [];
+
+    public function getDecisionType(string $name): string
```

Comment:
Here I would like us to have an "early return". And we should also validate that the value is a valid classname.

Reply:
Please make sure that we have 2 different exception messages for each use case to make DX better:

- "Decision for name ... not configured"
- "Decision for name ... not a valid classname"

Reply:
I have added check for valid classname and made a separate test for that. I'm not sure how we can use "early return" here though.